### PR TITLE
Created Story8/200Return Student/404Return newStudent from auth0 jwt

### PR DESCRIPTION
**Pull request by ENoll78**
Comments welcome.

### CODE REVIEW

File: src/main/java/com/collegecompendium/backend/controllers/StudentController.java [hunk 1]
```diff
@@ -44,6 +46,19 @@ public Student getStudentById(@PathVariable String id, HttpServletResponse respo
 		return query.get();
 	}
 	
+	@GetMapping("/student")
+	public Student getCallingStudent(@AuthenticationPrincipal Jwt token, HttpServletResponse response) {
+		Student result = studentRepository.findDistinctByAuth0Id(token.getSubject());
```
Comment (meta1203) [inline src/main/java/com/collegecompendium/backend/controllers/StudentController.java:51]: `findDistinctByAuth0Id` does not exist on the StudentRepository.

File: src/main/java/com/collegecompendium/backend/models/User.java [hunk 2]
```diff
@@ -72,32 +74,8 @@ public static HashMap<String, String> getAuth0(Jwt token) {
...
-	public void setLocation(String location) {
-		this.location = location;
-	}
-	public String getId() {
-		return id;
-	}
-	
+		
 	public void setId(String val) {
 		this.id = val;
```
Comment (meta1203) [inline src/main/java/com/collegecompendium/backend/models/User.java]: Is this still needed?

**Review by meta1203 — CHANGES_REQUESTED** [review]
Fix `findDistinctByAuth0Id` and maybe take a look at Student's left-over setter.

File: src/main/java/com/collegecompendium/backend/models/User.java (hunk 2 above)
Comment (ENoll78) [reply src/main/java/com/collegecompendium/backend/models/User.java]: We have it set id to Null in new student creation

File: src/main/java/com/collegecompendium/backend/controllers/StudentController.java (hunk 1 above)
Comment (ENoll78) [reply src/main/java/com/collegecompendium/backend/controllers/StudentController.java:51]: I have no idea why that didn't flag on my desktop?

**Review by meta1203 — APPROVED** [review]

**Pull request merged**